hopsworks-common/src/main/java/io/hops/hopsworks/common/serving/inference/InferenceController.java

```diff
@@ -33,39 +35,71 @@
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
+import static io.hops.hopsworks.common.serving.LocalhostServingController.PID_STOPPED;
+
+/**
+ * Contains the common functionality between serving instances for doing inference, delegates type-specific
+ * functionality to specific inference controllers like TfInferenceController, and SkLearnInferenceController
+ */
 @Stateless
 @TransactionAttribute(TransactionAttributeType.NEVER)
 public class InferenceController {
 
   private static final Logger logger = Logger.getLogger(InferenceLogger.class.getName());
 
   @EJB
-  private TfServingFacade tfServingFacade;
+  private ServingFacade servingFacade;
 
   @Inject
   private TfInferenceController tfInferenceController;
   @Inject
+  private SkLearnInferenceController skLearnInferenceController;
+  @Inject
   @Any
   private Instance<InferenceLogger> inferenceLoggers;
-
-
+  
+  
+  /**
+   * Makes an inference request to a running serving instance
+   *
+   * @param project the project where the serving is running
+   * @param modelName the name of the serving
+   * @param modelVersion the version of the serving
+   * @param verb the predictiont type (predict, regress, or classify)
+   * @param inferenceRequestJson the user-provided JSON payload for the inference request
+   * @return a string representation of the inference result
+   * @throws InferenceException
+   */
   public String infer(Project project, String modelName, Integer modelVersion,
                       String verb, String inferenceRequestJson) throws InferenceException {
 
-    TfServing tfServing = tfServingFacade.findByProjectModelName(project, modelName);
-    if (tfServing == null) {
+    Serving serving = servingFacade.findByProjectAndName(project, modelName);
+    if (serving == null) {
       throw new InferenceException(RESTCodes.InferenceErrorCode.SERVING_NOT_FOUND, Level.FINE, "name: " + modelName);
     }
+  
+    if (serving.getLocalPid().equals(PID_STOPPED)) {
```

Comment:
In the ee edition we don't have the pid - this class is used by both

Reply:
Done